### DISCUSSION TITLE
[Eyes]  License header check fails when the license is not Apache 2.0

### DISCUSSION THREAD

**Discussion by AltarIbnL** [apache/skywalking#9482] [Q&A] [answered]
I have referred to others' source files and tried to mark their license, it keeps showing that the file does not have a valid license header.
The license in my own project is Apache 2.0 and it passes the test. But the license of the new file is BSD-3-Clause and it keeps showing that the license is invalid.
Here is my .licenserc.yaml in root path:
![image](https://user-images.githubusercontent.com/33597692/185094136-87707703-1187-4649-b957-9ef60c32617f.png)

![image](https://user-images.githubusercontent.com/33597692/185094214-1242bd01-a9a0-413d-8d6b-508fcd6e0f6b.png)

It will failed while running  
`license-eye -c .licenserc.yaml  header check`

![image](https://user-images.githubusercontent.com/33597692/185094961-95fdac0d-185e-4dd0-bc17-c733b897b915.png)

However， if I change the license to Apache-2.0, It will pass the check successfully.
![image](https://user-images.githubusercontent.com/33597692/185095395-0c008398-04cf-4d3a-82b0-7eb7303f560a.png)

![image](https://user-images.githubusercontent.com/33597692/185095418-622bb93e-7225-4440-adc3-c9d688fefe49.png)

Is there something wrong with my steps, sorry for bordering you.

**Comment by wu-sheng** [comment]
Ping @kezhenxu94

**Comment by kezhenxu94** [comment, marked as answer]
Hi. Why do you specify three different licenses for the same `path`?

**Reply by AltarIbnL** [reply]
Hi. In my project, we use Apache 2.0 as our license while the external library (Kubernetes, Mockery) we referenced is Apache 2.0 and BSD-3-Clause. We specify three licenses on the same path due to the files are on the same path. 
Mockery will generate a template code automatically, which should tag with their license. The problem we met is no matter what license we set in `.licenserc.yaml `, it will fail with an invalid license even if we change the path to the package it's located.

![image](https://user-images.githubusercontent.com/33597692/185161135-43fcf133-0863-4e2e-b8a3-d227de304838.png)

**Reply by kezhenxu94** [reply]
> it will fail with an invalid license even if we change the path to the package it's located.
> 

Please add the path to `paths-ignore` in the other two license section.

**Reply by AltarIbnL** [reply]
` Please add the path to paths-ignore in the other two license section.`
Thx! The problem fixed after I add  the path to paths-ignor.
